apps/remix-ide/src/app/tabs/runTab/settings.js

```diff
@@ -244,16 +244,29 @@ class SettingsUI {
   }
 
   setExecutionContext (context) {
-    this.blockchain.changeExecutionContext(context, () => {
-      modalDialogCustom.prompt('External node request', this.web3ProviderDialogBody(), 'http://127.0.0.1:8545', (target) => {
-        this.blockchain.setProviderFromEndpoint(target, context, (alertMsg) => {
-          if (alertMsg) addTooltip(alertMsg)
-          this.setFinalContext()
-        })
+    if (context === 'Hardhat Provider') {
+      this.blockchain.changeExecutionContext(context, () => {
+        modalDialogCustom.prompt('Hardhat node request', this.hardhatProviderDialogBody(), 'http://127.0.0.1:8545', (target) => {
```

Comment:
wasn't it possible to put this in https://github.com/ethereum/remix-project/pull/1206/files#diff-aedcee9b7a11a19d2ec1b9a1831bd5d4985392e3701b74003c0171f1b2e74c2bR21 ?
I mean your way work but it'd be better to have hardhat stuff in a single.
Also having declared a whole plugin but still need to do that is a bit overkill.

Reply:
done